### PR TITLE
Fix: Added Sentiment Analysis Bengali Dataset

### PR DESCRIPTION
**Pull request by Akash190104**
Added Sentiment Analysis Dataset in Bengali
## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

**Review by asparius — COMMENTED** [review]
Looks good, but unsure of usage of train as an eval split, we could only have test split instead. More of nomenclature thing, so @KennethEnevoldsen or @isaac-chung can tell more.

File: mteb/tasks/Classification/ben/SentimentAnalysisBengali.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        pages={50--60},
+        year={2020}
+        }""",
+        n_samples={"train": 11807},
+        avg_character_length={"train": 69.66},
+    )
+
+    def dataset_transform(self):
+        N_SAMPLES = 2048
+        self.dataset["train"] = self.dataset["train"].select(range(N_SAMPLES))
```
Comment (isaac-chung) [inline mteb/tasks/Classification/ben/SentimentAnalysisBengali.py:43]:
```suggestion
        self.dataset = self.stratified_subsampling(
            self.dataset, seed=self.seed, splits=["train"]
        )
```

File: mteb/tasks/Classification/ben/SentimentAnalysisBengali.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+from mteb.abstasks.AbsTaskClassification import AbsTaskClassification
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+
+class SentimentAnalysisBengali(AbsTaskClassification):
```
Comment (isaac-chung) [inline mteb/tasks/Classification/ben/SentimentAnalysisBengali.py:7]: Please update all references as well, including the model runs.
```suggestion
class BengaliSentimentAnalysis(AbsTaskClassification):
```

**Review by isaac-chung — CHANGES_REQUESTED** [review]
Thanks for your contributions. Just a few comments.

**Comment by isaac-chung** [comment]
> Looks good, but unsure of usage of train as an eval split, we could only have test split instead. More of nomenclature thing, so @KennethEnevoldsen or @isaac-chung can tell more.

Should be fine. Datasets like `HotelReviewSentimentClassification` are in the same situation.

**Review by isaac-chung — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**